### ISSUE TITLE
Unable to switch branch

### ISSUE THREAD

**Issue by wrighke** [desktop/desktop#14300]
### The problem

After creating a second branch I am unable to switch back into my original branch. 
Upon trying to switch branches I am prompted to stash changes and continue, which I did. But then nothing happens and it stays on the same branch.
Upon trying this a few times there is an error in the logs that the repository is not found. Has my work been deleted?
Thanks

### Release version

Version 2.9.0 (x64)

### Operating system

Windows 10 Version 20H2 (OS Build 19032.1586)

### Steps to reproduce the behavior

1. create branch
2. create second branch
3. switch to second branch, choosing to leave changes on current branch
4. attempt to change back to first branch

### Log files

(The error was parsed as 37: Unable to switch branches as there are working directory changes which would be overwritten. Please commit or stash your changes.)
2022-04-07T04:40:56.906Z - info: [ui] Executing getStatus: git --no-optional-locks status --untracked-files=all --branch --porcelain=2 -z (took 3.345s)
2022-04-07T04:40:59.236Z - info: [ui] [Timing] Action 'stash and checkout' for 'dea/HFR_Repository' took 21.740s
2022-04-07T04:41:06.924Z - info: [ui] Executing getRemotes: git remote -v (took 1.831s)
2022-04-07T04:41:08.021Z - info: [ui] Executing getStatus: git --no-optional-locks status --untracked-files=all --branch --porcelain=2 -z (took 3.812s)
2022-04-07T04:42:03.282Z - info: [ui] Executing getStatus: git --no-optional-locks status --untracked-files=all --branch --porcelain=2 -z (took 1.466s)
2022-04-07T04:42:05.232Z - info: [ui] Executing getRemotes: git remote -v (took 1.947s)
2022-04-07T04:42:05.547Z - info: [ui] [AppStore.withAuthenticatingUser] account found for repository: logit-common-ds-apps - wrighke (has token)
2022-04-07T04:42:06.748Z - info: [ui] Executing createStashEntry: git stash push -m !!GitHub_Desktop<main> (took 4.504s)
2022-04-07T04:42:06.860Z - info: [ui] Executing getStatus: git --no-optional-locks status --untracked-files=all --branch --porcelain=2 -z (took 5.059s)
2022-04-07T04:42:09.576Z - info: [ui] Executing fetch: git -c credential.helper= fetch --progress --prune origin (took 4.014s)
2022-04-07T04:42:23.844Z - info: [ui] Executing getStatus: git --no-optional-locks status --untracked-files=all --branch --porcelain=2 -z (took 2.310s)
2022-04-07T04:44:02.190Z - info: [ui] Executing getStatus: git --no-optional-locks status --untracked-files=all --branch --porcelain=2 -z (took 1.504s)
2022-04-07T04:44:02.454Z - info: [ui] Executing getRemotes: git remote -v (took 1.522s)
2022-04-07T04:44:03.266Z - info: [ui] Executing getRemotes: git remote -v (took 1.074s)
2022-04-07T04:44:04.164Z - info: [ui] Executing getStatus: git --no-optional-locks status --untracked-files=all --branch --porcelain=2 -z (took 3.505s)
2022-04-07T04:44:11.156Z - warn: [ui] fetchRepository: 'wrighke/Files' returned a 404
2022-04-07T04:44:11.245Z - warn: [ui] fetchRepository: 'wrighke/Files' returned a 404
2022-04-07T04:44:11.245Z - info: [ui] [AppStore.withAuthenticatingUser] account found for repository: Files - wrighke (has token)
2022-04-07T04:44:12.248Z - error: [ui] `git -c credential.helper= fetch --progress --prune origin` exited with an unexpected code: 128.
stderr:
remote: Repository not found.
fatal: repository '<redacted>' not found

(The error was parsed as 8: The repository does not seem to exist anymore. You may not have access, or it may have been deleted or renamed.)

### Screenshots

![image](https://user-images.githubusercontent.com/103163606/162122003-8741f024-d1b1-4776-bedb-7a8ae4bd672e.png)
![image](https://user-images.githubusercontent.com/103163606/162122014-e4299c2b-d5c2-4ed4-ab7f-f11c3c968166.png)
![image](https://user-images.githubusercontent.com/103163606/162122039-fb43f4b1-bd19-4a84-8ad4-2241c47309f8.png)

**Comment by steveward** [comment]
Thanks for the report @wrighke. It looks like the error in the log file is for a different repository. That error indicates that the remote repository is unable to be found.

I'm not seeing any errors related to stashing -- what happens if you right click on `22 changed files` and select `Stash all changes`?

**Comment by wrighke** [comment]
Hi @steveward nothing happens when attempting to stash the changes on main branch. 
From the logs it looks like it just runs in a loop attempting to stash and then hits a 404 error:

2022-04-07T00:25:01.045Z - info: [ui] Executing createStashEntry: git stash push -m !!GitHub_Desktop<dataupload> (took 4.436s)
2022-04-07T00:25:01.285Z - info: [ui] [Timing] Action 'stash and checkout' for 'dea/HFR_Repository' took 15.342s
2022-04-07T00:25:15.137Z - info: [ui] [RepositoryIndicatorUpdater]: Refreshing sidebar indicators for 56 repositories took 67.9s of which 2640.8s paused, total 2708.7s
2022-04-07T00:25:16.665Z - info: [ui] [Timing] Action 'checkout branch from list' for 'dea/HFR_Repository' took 0.000s
2022-04-07T00:25:18.214Z - info: [ui] [AppStore.withAuthenticatingUser] account found for repository: HFR_Repository - wrighke (has token)
2022-04-07T00:25:21.448Z - info: [ui] Executing createStashEntry: git stash push -m !!GitHub_Desktop<dataupload> (took 2.903s)
2022-04-07T00:25:25.050Z - info: [ui] Executing checkoutBranch: git -c credential.helper= checkout --progress IPNet -- (took 3.581s)
2022-04-07T00:25:25.053Z - error: [ui] `git -c credential.helper= checkout --progress IPNet --` exited with an unexpected code: 1.
stderr:
error: Your local changes to the following files would be overwritten by checkout:
`the 22 files which needed to be stashed`
Aborting

(The error was parsed as 37: Unable to switch branches as there are working directory changes which would be overwritten. Please commit or stash your changes.)
2022-04-07T00:25:28.057Z - info: [ui] Executing getStatus: git --no-optional-locks status --untracked-files=all --branch --porcelain=2 -z (took 2.336s)
2022-04-07T00:25:29.737Z - info: [ui] Executing getStatus: git --no-optional-locks status --untracked-files=all --branch --porcelain=2 -z (took 4.681s)
2022-04-07T00:25:30.868Z - info: [ui] Executing getStatus: git --no-optional-locks status --untracked-files=all --branch --porcelain=2 -z (took 2.157s)
2022-04-07T00:25:31.416Z - info: [ui] Executing getBranches: git for-each-ref --format=%00%(refname)%00%(refname:short)%00%(upstream:short)%00%(objectname)%00%(author)%00%(symref)%00 refs/heads refs/remotes (took 1.546s)
2022-04-07T00:25:31.468Z - info: [ui] Executing getRecentBranches: git log -g --no-abbrev-commit --pretty=oneline HEAD -n 2500 -- (took 1.468s)
2022-04-07T00:25:31.737Z - info: [ui] Executing getWorkingDirectoryDiff: git diff --no-ext-diff --patch-with-raw -z --no-color HEAD -- Instances/CNS-enums/repositories/CNS-GA-Prod-Splunk-HForwarder-master/etc/apps/TA-Github/bin/ta_github/lib/python2.7/site-packages/github/tests/ReplayData/AuthenticatedUser.testCreateAuthorizationWithAllArguments.txt (took 1.999s)
2022-04-07T00:25:32.307Z - info: [ui] Executing createStashEntry: git stash push -m !!GitHub_Desktop<dataupload> (took 3.973s)
2022-04-07T00:25:33.179Z - info: [ui] Executing getCommits: git log HEAD --date=raw --max-count=100 -z --format=%H%x00%h%x00%s%x00%b%x00%an <%ae> %ad%x00%cn <%ce> %cd%x00%P%x00%(trailers:unfold,only)%x00%D --no-show-signature --no-color --not --remotes -- (took 1.203s)
2022-04-07T00:25:33.273Z - info: [ui] Executing getRemotes: git remote -v (took 1.269s)
2022-04-07T00:25:34.412Z - info: [ui] [Timing] Action 'stash and checkout' for 'dea/HFR_Repository' took 16.193s
2022-04-07T00:25:39.736Z - info: [ui] Executing getRemotes: git remote -v (took 1.268s)
2022-04-07T00:25:41.834Z - info: [ui] Executing getStatus: git --no-optional-locks status --untracked-files=all --branch --porcelain=2 -z (took 1.855s)
2022-04-07T00:25:42.355Z - info: [ui] Executing createStashEntry: git stash push -m !!GitHub_Desktop<dataupload> (took 3.835s)
2022-04-07T00:25:50.150Z - info: [ui] Executing getRemotes: git remote -v (took 1.070s)
2022-04-07T00:25:51.506Z - info: [ui] Executing getStatus: git --no-optional-locks status --untracked-files=all --branch --porcelain=2 -z (took 1.066s)
2022-04-07T00:25:51.947Z - info: [ui] [Timing] Action 'checkout branch from list' for 'dea/HFR_Repository' took 0.000s
2022-04-07T00:25:52.031Z - info: [ui] Executing createStashEntry: git stash push -m !!GitHub_Desktop<dataupload> (took 2.906s)
2022-04-07T00:25:53.837Z - info: [ui] [AppStore.withAuthenticatingUser] account found for repository: HFR_Repository - wrighke (has token)
2022-04-07T00:25:56.281Z - info: [ui] Executing getStatus: git --no-optional-locks status --untracked-files=all --branch --porcelain=2 -z (took 1.758s)
2022-04-07T00:25:56.777Z - warn: [ui] fetchRepository: 'wrighke/Files' returned a 404
2022-04-07T00:25:56.855Z - warn: [ui] fetchRepository: 'wrighke/Files' returned a 404
2022-04-07T00:25:56.856Z - info: [ui] [AppStore.withAuthenticatingUser] account found for repository: Files - wrighke (has token)
2022-04-07T00:25:56.935Z - info: [ui] Executing createStashEntry: git stash push -m !!GitHub_Desktop<dataupload> (took 2.781s)
2022-04-07T00:25:59.353Z - info: [ui] Executing fetch: git -c credential.helper= fetch --progress --prune origin (took 2.472s)
2022-04-07T00:25:59.354Z - error: [ui] `git -c credential.helper= fetch --progress --prune origin` exited with an unexpected code: 128.
stderr:
remote: Repository not found.
fatal: repository `companyURL`'/wrighke/Files.git/' not found

(The error was parsed as 8: The repository does not seem to exist anymore. You may not have access, or it may have been deleted or renamed.)

**Comment by wrighke** [comment]
FYI My work is still stashed:
![image](https://user-images.githubusercontent.com/103163606/162339419-b1d93efd-2a39-4f1e-b147-869b1f4303d3.png)

**Comment by wrighke** [comment]
Fixed the issue. When I tried to manually discard the 22 changed files in `main` branch, it was revealed the filenames were too long for Windows to handle. For some reason this error is not logged in the Github Desktop logs. 
I simply ran `git config --global core.longpaths true` to allow long filenames, then discard the 22 changed files with `git checkout -- .`, then I was able to simply switch back to the other branch and restore the stashed changes.
